src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -404,30 +405,77 @@ public async Task<List<GitFile>> GetFilesAtCommitAsync(string repoUri, string co
                     .Select(
                         async treeItem =>
                         {
-                            Blob blob = await ExponentialRetry.RetryAsync(
+                            return await GetGitTreeItem(path, treeItem, owner, repo);
+                        }));
+            return files.ToList();
+        }
+
+        /// <summary>
+        ///     Get a tree item blob from github, using the cache if it exists.
+        /// </summary>
+        /// <param name="path">Base path of final git file</param>
+        /// <param name="treeItem">Tree item to retrieve</param>
+        /// <param name="owner">Organization</param>
+        /// <param name="repo">Repository</param>
+        /// <returns>Git file with tree item contents.</returns>
+        public async Task<GitFile> GetGitTreeItem(string path, TreeItem treeItem, string owner, string repo)
+        {
+            // If we have a cache available here, attempt to get the value in the cache
+            // before making the request. Generally, we are requesting the same files for each
+            // arcade update sent to the each repository daily per subscription. This is inefficient, as it means we
+            // request each file N times, where N is the number of subscriptions. The number of files (M),
+            // is non-trivial, so reducing N*M to M is vast improvement.
+            // Use the treeItem.Sha as the key. I think it is overkill to hash the repo and owner into
+            // the key.
+
+            if (Cache != null)
+            {
+                return await Cache.GetOrCreateAsync(treeItem.Sha, async (entry) =>
+                {
+                    return await GetGitItemImpl(path, treeItem, owner, repo);
```

Comment:
Reading up on the docs for the MemoryCache, I see this section: https://dev.azure.com/dnceng/internal/_build/results?buildId=302513&view=results

```The memory size limit does not have a defined unit of measure because the cache has no mechanism to measure the size of entries. If the cache memory size limit is set, all entries must specify size. The ASP.NET Core runtime does not limit cache size based on memory pressure. It's up to the developer to limit cache size. The size specified is in units the developer chooses.```

Looks like we should add the file size to the entry, or the size option for the cache won't actually do what we hope to cap at 64MB?

Reply:
Ooh good call.

Reply:
FIxed.